### PR TITLE
Rebase Frontend with main branch

### PR DESCRIPTION
**Pull request by hardikpatil**
This PR is to merge all the feature updates from `frontend` to `main`.
- Feed page content @hardikpatil 
- Donor page content @kushalramakanth @decoles 
- Redux state management @xl26 
- Registration success page @prasannarajezzzy 
- Sale page category icons @hardikpatil @AbhinavReddy-Dev 
- Sale page table @DevangRaval1 @prasannarajezzzy @TJSidhu1  | @arshiya19 @Gajjela-Koushik-Reddy @KishoreMenda 
- Formik update for login form @priyankavrao97 
- Formik update for Register form @SANDESHSOBARAD

### CODE REVIEW

**Comment by kbuffardi** [comment]
@hardikpatil Ideally, we would have had smaller individual issues for each of these features.

However, it would be more work than it's worth to separate them all at this stage. If the reviews approve the PR, let's Merge Commit so it retains the separate commits of each feature.

**Comment by AbhinavReddy-Dev** [comment]
> @hardikpatil Ideally, we would have had smaller individual issues for each of these features.
> 
> However, it would be more work than it's worth to separate them all at this stage. If the reviews approve the PR, let's Merge Commit so it retains the separate commits of each feature.

I totally agree with you on the point of smaller individual issues, after the feature branches were squash merged into frontend with mvp changes initially, there were no immediate PRs to `main` from `frontend`, hence the million commits.

File: PantryNodeReact/public/index.html
Comment (AbhinavReddy-Dev) [inline PantryNodeReact/public/index.html:27]: I'm not sure who made these changes and why, it is not supposed to be branded as Chico State. Need to remove "Chico State" from the title.

File: PantryNodeReact/src/Components/Feed/Feed.tsx
Comment (AbhinavReddy-Dev) [inline PantryNodeReact/src/Components/Feed/Feed.tsx:39]: We can also just return isExpiring and remove the return null.

File: PantryNodeReact/src/Components/Sale/SaleTable.tsx
Comment (AbhinavReddy-Dev) [inline PantryNodeReact/src/Components/Sale/SaleTable.tsx:27]: This should be changed from type to interface.

File: PantryNodeReact/src/Components/Sale/SaleTable.tsx
Comment (AbhinavReddy-Dev) [inline PantryNodeReact/src/Components/Sale/SaleTable.tsx:33]: This should be changed from type to interface.

**Review by AbhinavReddy-Dev — DISMISSED** [review]
All the changes requested can be a clean up commit to `main` along with upcoming and current PRs to `frontend` in a couple days.

File: PantryNodeReact/src/Components/Feed/Feed.tsx
Comment (hardikpatil) [reply PantryNodeReact/src/Components/Feed/Feed.tsx:39]: correct

File: PantryNodeReact/public/index.html
Comment (hardikpatil) [reply PantryNodeReact/public/index.html]: changed it to `Food Pantry` for now

**Review by kushalramakanth — APPROVED** [review]
LGTM!

**Review by kushalramakanth — APPROVED** [review]
LGTM, but after registration successfully, login redirect is not working. Need to add new issue for that.

**Comment by hardikpatil** [comment]
> LGTM, but after registration successfully, login redirect is not working. Need to add new issue for that.

I tested on both, locally and on codespace (deleting previous instances) , redirection working fine on me. Maybe, your codespace might have cached. Thanks.

**Comment by AbhinavReddy-Dev** [comment]
There seems to be a conflict with the newly added login api connection and Formik update in login form, need someone to take up this issue and try to resolve it and make it work as expected. I believe this will count towards an important commit towards `main`.

**Comment by DevangRaval1** [comment]
ReviewRebase

Everything Works except the login which as discussed we need to create a separate issue for.

**Review by DevangRaval1 — APPROVED** [review]
LGTM!

**Review by kushalramakanth — APPROVED** [review]
LGTM!

**Comment by kbuffardi** [comment]
I believe this PR is causing errors. When I pulled the latest from `main` and ran the container with `docker-compose build` and `docker-compose up` I receive the following errors:

```
react                  | Failed to compile.
react                  | 
react                  | Module not found: Error: Can't resolve 'react-redux' in '/app/frontend/src'
react                  | ERROR in ./src/hooks.ts 3:0-55
react                  | Module not found: Error: Can't resolve 'react-redux' in '/app/frontend/src'
react                  | 
react                  | ERROR in ./src/index.tsx 17:0-39
react                  | Module not found: Error: Can't resolve 'react-redux' in '/app/frontend/src'
react                  | 
react                  | ERROR in ./src/pages/login.tsx 24:0-35
react                  | Module not found: Error: Can't resolve 'formik' in '/app/frontend/src/pages'
react                  | 
react                  | ERROR in ./src/pages/register.tsx 25:0-35
react                  | Module not found: Error: Can't resolve 'formik' in '/app/frontend/src/pages'
react                  | 
react                  | ERROR in ./src/redux-features/user.ts 3:0-65
react                  | Module not found: Error: Can't resolve '@reduxjs/toolkit' in '/app/frontend/src/redux-features'
react                  | 
react                  | ERROR in ./src/store.ts 3:0-50
react                  | Module not found: Error: Can't resolve '@reduxjs/toolkit' in '/app/frontend/src'
react                  | 
react                  | webpack compiled with 6 errors
react                  | ERROR in src/hooks.ts:1:42
react                  | TS2307: Cannot find module 'react-redux' or its corresponding type declarations.
react                  |   > 1 | import { useDispatch, useSelector } from 'react-redux'
react                  |       |                                          ^^^^^^^^^^^^^
react                  |     2 | import type { TypedUseSelectorHook } from 'react-redux'
react                  |     3 | import type { RootState, AppDispatch } from './store'
react                  |     4 |
react                  | 
react                  | ERROR in src/hooks.ts:2:43
react                  | TS2307: Cannot find module 'react-redux' or its corresponding type declarations.
react                  |     1 | import { useDispatch, useSelector } from 'react-redux'
react                  |   > 2 | import type { TypedUseSelectorHook } from 'react-redux'
react                  |       |                                           ^^^^^^^^^^^^^
react                  |     3 | import type { RootState, AppDispatch } from './store'
react                  |     4 |
react                  |     5 | // Use throughout your app instead of plain `useDispatch` and `useSelector`
react                  | 
react                  | ERROR in src/index.tsx:15:26
react                  | TS2307: Cannot find module 'react-redux' or its corresponding type declarations.
react                  |     13 |
react                  |     14 | import { store } from "./store";
react                  |   > 15 | import { Provider } from 'react-redux'
react                  |        |                          ^^^^^^^^^^^^^
react                  |     16 |
react                  |     17 | const rootElement = document.getElementById("root");
react                  |     18 | const root = ReactDOM.createRoot(rootElement!);
react                  | 
react                  | ERROR in src/pages/login.tsx:22:27
react                  | TS2307: Cannot find module 'formik' or its corresponding type declarations.
react                  |     20 |
react                  |     21 | import { useState } from "react";
react                  |   > 22 | import { useFormik } from "formik";
react                  |        |                           ^^^^^^^^
react                  |     23 | import axios from 'axios';
react                  |     24 |
react                  |     25 | type LoginFormInput = {
react                  | 
react                  | ERROR in src/pages/register.tsx:21:27
react                  | TS2307: Cannot find module 'formik' or its corresponding type declarations.
react                  |     19 | import { useState } from "react";
react                  |     20 | import Success from "./success";
react                  |   > 21 | import { useFormik } from "formik";
react                  |        |                           ^^^^^^^^
react                  |     22 |
react                  |     23 | type RegisterFormInput = {
react                  |     24 |   firstName?: string;
react                  | 
react                  | ERROR in src/redux-features/user.ts:1:47
react                  | TS2307: Cannot find module '@reduxjs/toolkit' or its corresponding type declarations.
react                  |   > 1 | import { createSlice, createAsyncThunk } from '@reduxjs/toolkit'
react                  |       |                                               ^^^^^^^^^^^^^^^^^^
react                  |     2 | //import { PayloadAction } from '@reduxjs/toolkit'
react                  |     3 | import axiosInstance from '../util/axiosInstance'
react                  |     4 |
react                  | 
react                  | ERROR in src/store.ts:1:32
react                  | TS2307: Cannot find module '@reduxjs/toolkit' or its corresponding type declarations.
react                  |   > 1 | import { configureStore } from '@reduxjs/toolkit'
react                  |       |                                ^^^^^^^^^^^^^^^^^^
react                  |     2 | import userReducer from './redux-features/user'
react                  |     3 |
react                  |     4 | export const store = configureStore({
```

**Comment by hardikpatil** [comment]
> I believe this PR is causing errors. When I pulled the latest from `main` and ran the container with `docker-compose build` and `docker-compose up` I receive the following errors:
> 
> ```
> react                  | Failed to compile.
> react                  | 
> react                  | Module not found: Error: Can't resolve 'react-redux' in '/app/frontend/src'
> react                  | ERROR in ./src/hooks.ts 3:0-55
> react                  | Module not found: Error: Can't resolve 'react-redux' in '/app/frontend/src'
> react                  | 
> react                  | ERROR in ./src/index.tsx 17:0-39
> react                  | Module not found: Error: Can't resolve 'react-redux' in '/app/frontend/src'
> react                  | 
> react                  | ERROR in ./src/pages/login.tsx 24:0-35
> react                  | Module not found: Error: Can't resolve 'formik' in '/app/frontend/src/pages'
> react                  | 
> react                  | ERROR in ./src/pages/register.tsx 25:0-35
> react                  | Module not found: Error: Can't resolve 'formik' in '/app/frontend/src/pages'
> react                  | 
> react                  | ERROR in ./src/redux-features/user.ts 3:0-65
> react                  | Module not found: Error: Can't resolve '@reduxjs/toolkit' in '/app/frontend/src/redux-features'
> react                  | 
> react                  | ERROR in ./src/store.ts 3:0-50
> react                  | Module not found: Error: Can't resolve '@reduxjs/toolkit' in '/app/frontend/src'
> react                  | 
> react                  | webpack compiled with 6 errors
> react                  | ERROR in src/hooks.ts:1:42
> react                  | TS2307: Cannot find module 'react-redux' or its corresponding type declarations.
> react                  |   > 1 | import { useDispatch, useSelector } from 'react-redux'
> react                  |       |                                          ^^^^^^^^^^^^^
> react                  |     2 | import type { TypedUseSelectorHook } from 'react-redux'
> react                  |     3 | import type { RootState, AppDispatch } from './store'
> react                  |     4 |
> react                  | 
> react                  | ERROR in src/hooks.ts:2:43
> react                  | TS2307: Cannot find module 'react-redux' or its corresponding type declarations.
> react                  |     1 | import { useDispatch, useSelector } from 'react-redux'
> react                  |   > 2 | import type { TypedUseSelectorHook } from 'react-redux'
> react                  |       |                                           ^^^^^^^^^^^^^
> react                  |     3 | import type { RootState, AppDispatch } from './store'
> react                  |     4 |
> react                  |     5 | // Use throughout your app instead of plain `useDispatch` and `useSelector`
> react                  | 
> react                  | ERROR in src/index.tsx:15:26
> react                  | TS2307: Cannot find module 'react-redux' or its corresponding type declarations.
> react                  |     13 |
> react                  |     14 | import { store } from "./store";
> react                  |   > 15 | import { Provider } from 'react-redux'
> react                  |        |                          ^^^^^^^^^^^^^
> react                  |     16 |
> react                  |     17 | const rootElement = document.getElementById("root");
> react                  |     18 | const root = ReactDOM.createRoot(rootElement!);
> react                  | 
> react                  | ERROR in src/pages/login.tsx:22:27
> react                  | TS2307: Cannot find module 'formik' or its corresponding type declarations.
> react                  |     20 |
> react                  |     21 | import { useState } from "react";
> react                  |   > 22 | import { useFormik } from "formik";
> react                  |        |                           ^^^^^^^^
> react                  |     23 | import axios from 'axios';
> react                  |     24 |
> react                  |     25 | type LoginFormInput = {
> react                  | 
> react                  | ERROR in src/pages/register.tsx:21:27
> react                  | TS2307: Cannot find module 'formik' or its corresponding type declarations.
> react                  |     19 | import { useState } from "react";
> react                  |     20 | import Success from "./success";
> react                  |   > 21 | import { useFormik } from "formik";
> react                  |        |                           ^^^^^^^^
> react                  |     22 |
> react                  |     23 | type RegisterFormInput = {
> react                  |     24 |   firstName?: string;
> react                  | 
> react                  | ERROR in src/redux-features/user.ts:1:47
> react                  | TS2307: Cannot find module '@reduxjs/toolkit' or its corresponding type declarations.
> react                  |   > 1 | import { createSlice, createAsyncThunk } from '@reduxjs/toolkit'
> react                  |       |                                               ^^^^^^^^^^^^^^^^^^
> react                  |     2 | //import { PayloadAction } from '@reduxjs/toolkit'
> react                  |     3 | import axiosInstance from '../util/axiosInstance'
> react                  |     4 |
> react                  | 
> react                  | ERROR in src/store.ts:1:32
> react                  | TS2307: Cannot find module '@reduxjs/toolkit' or its corresponding type declarations.
> react                  |   > 1 | import { configureStore } from '@reduxjs/toolkit'
> react                  |       |                                ^^^^^^^^^^^^^^^^^^
> react                  |     2 | import userReducer from './redux-features/user'
> react                  |     3 |
> react                  |     4 | export const store = configureStore({
> ```

@kbuffardi not sure, however, I don't think any of the frontend team touched docker-related file, we normally test it out on `cd PantryNodeReact` with `npm install` and `npm start` . I am getting the below message when I did `docker-compose build` and `docker-compse up` in the root directory.  @AbhinavReddy-Dev any clue what this might be?

Screenshot 2023-04-12 at 11 59 46

**Comment by kbuffardi** [comment]
Let me clarify that I don't think the docker configurations are the problem, but I think the updates don't work within the container... and that is a problem. If you run things locally while you develop that's fine, but it's a **necessity** to confirm that any PRs work within the team's DevOps (including docker container and CI) without breaking. This should be:

* tested by the dev before making the PR
* tested by each reviewer and not approved until it is confirmed that it will work

**Comment by AbhinavReddy-Dev** [comment]
The issue arises when the the app is trying to access the newly added packages, for the frontend it is looking in the `app/frontend/src` folder instead of `app/frontend/node_modules`. I think it is the same issue with backend when accessing cors package. Need to resolve this ASAP.

**Comment by AbhinavReddy-Dev** [comment]
Created a new issue for this: https://github.com/ChicoState/PantryNode/issues/139

**Pull request merged**
